sentry-actix/src/lib.rs

```diff
@@ -351,7 +351,9 @@ where
             let mut res: Self::Response = match fut.await {
                 Ok(res) => res,
                 Err(e) => {
-                    if inner.capture_server_errors {
+                    // Errors returned by middleware, and possibly other lower level errors
+                    if inner.capture_server_errors && e.error_response().status().is_server_error()
```

Comment:
[future improvement] Does the Rust SDK offer a way to configure the exact status codes which are sent to Sentry? We have such a feature in several of the Python SDK's server integrations; maybe it would be worth it to consider adding this feature to the Rust SDK, as well.